core/modules/filters/qfp.js

```diff
@@ -0,0 +1,40 @@
+/*\
+title: $:/core/modules/filters/qfp.js
+type: application/javascript
+module-type: filteroperator
+
+Filter operator that returns a "qualified" hashstring for a list of input titles
+Example feeding it with a list of transclusion variables that model the path
+from a given point up to the page template (returns the same qualified string
+like the <<qualify>> macro would do at that point):
+
+filter="{|$:/core/ui/PageTemplate/story|||} {|$:/core/ui/PageTemplate|||} +[qfp[]]"
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+/*
+Export our filter function
+*/
+exports.qfp = function(source,operator,options) {
+	var titles = [],
+	results = [],
+	hashString;
+	source(function(tiddler,title) {
+			titles.push(title);
+	});
+	if(titles.length) {
+		hashString = $tw.utils.hashString(titles.join("")).toString();
+		while(hashString.length < 11) {
```

Comment:
Presumably the padding is to make sure that every hash is the same length? Any reason for padding with dashes?

Reply:
the reason for padding with dashes is to get the same string that the `<<qualify>>` macro would produce